### ISSUE TITLE
False positive StringLiteralDuplication

### ISSUE THREAD

**Issue by Dosexe** [detekt/detekt#8074]
## Expected Behavior

Rule `StringLiteralDuplication` should not be triggered

## Observed Behavior

Code triggers `StringLiteralDuplication` rule

## Steps to Reproduce

run `gradle detekt` over code below
```kt
object ConfigUtils {

    private val prefix = "test"

    fun createX(name: String): String = "$prefix$name"

    fun createY(name: String): String = "$prefix$name"
}
```

## Context

In my example it is clear that the strings are not the same, since the logic under different function can be different and the `name` param may mean different things and have different values in the context. If code is replaced with `prefix + name` everything works as expected, but this is undesired code style.

Rule setup
```
  StringLiteralDuplication:
    active: true
    excludes: ['**/test/**', '**/androidTest/**', '**/commonTest/**', '**/jvmTest/**', '**/jsTest/**', '**/iosTest/**']
    threshold: 2
    ignoreAnnotation: true
    excludeStringsWithLessThan5Characters: true
    ignoreStringsRegex: '$^'
```

## Your Environment

* Version of detekt used:  `1.23.8`
* Version of Gradle used (if applicable):  `8.12`
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: 
* Link to your project (if it's a public repository):

**Comment by markoeltiger** [comment]
can i get assigned for this one?

**Comment by BraisGabin** [comment]
I'm not 100% sure how this can be fixed. Where do we draw the line? If that string is a really long string that only append at the end $name. Should we still doesn't flag it? In that case I would say that probably a private function that creates the string would be better than this duplication. So in that case the report would be valid. 

If we can define a clear line to when is duplication and when it is not would be great. Or should we just add a flag to ignore all strings with interpolation? Would that fix the issue for you @Dosexe?

To be honest, I don't like this rule too much because the duplication some times have sense and forbid it can harm more than heal.

**Comment by Dosexe** [comment]
@BraisGabin I would say that ignore strings with interpolation is a good enough solution.

**Comment by BraisGabin** [comment]
@detekt/maintainers I would like another opinion here. Should we just ignore the ones with placeholders, add a new flag or keep the rule as it is right now?

**Comment by schalkms** [comment]
👍
> ignore the ones with placeholders

**Comment by TWiStErRob** [comment]
I think ignoring with placeholders is not in line with the rule's intent.

In the OP example there are already two createX and createY functions with the same body, so that is being highlighted, why not have createZ that covers both?

I think OPs example is an edge case though, there are other cases where duplication with placeholders can be seen as proper dupes, for example throwing the same exception with message and placeholders, clear candidate for factory/throw function.

There's something to be considered though where extracting a function that builds a change complex string with placeholders is not possible: different types. It would probably make sense if the placeholder containing strings would be keyed together with their types, for example: `val bar = 1; "foo$bar"` would keyed as "foo$Int" and only flagged if there's another few occurrences of similar interpolations, even if variable names are different, or a `${}` evaluates to the same type.

If this is too complicated I think a flag with a default that's the same as current behaviour would be a nice middle ground.

Overall I think that the OP example is a case of "this is fine" and a suppression is the way to go, rather than blanket ignoring all interpolations in the codebase.

**Comment by BraisGabin** [comment]
It looks that there are different opinions here. So, I guess, the best option is to add a flag to allow to configure it.

**Comment by Dosexe** [comment]
@BraisGabin Yes, sounds like the best option here.

**Comment by BraisGabin** [comment]
@Dosexe do you want to give it a try and send a PR?
